kotlin-codegen/runtime/src/main/java/com/squareup/moshi/MoshiSerializable.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi
+
+import java.lang.reflect.Constructor
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+import kotlin.annotation.AnnotationRetention.RUNTIME
+import kotlin.annotation.AnnotationTarget.CLASS
+
+@Retention(RUNTIME)
+@Target(CLASS)
+annotation class MoshiSerializable
+
+class MoshiSerializableFactory : JsonAdapter.Factory {
```

Comment:
Why not make this an `object`?

Reply:
I asked higher up about whether or not to make this class a singleton (lazy or otherwise), and we went with just making this instantiable

Reply:
It’s consistent with what we did for reflection.
KotlinJsonAdapterFactory